effects/cats/src/test/scala/sttp/client3/impl/cats/CatsTestBase.scala

```diff
@@ -1,21 +1,20 @@
 package sttp.client3.impl.cats
 
-import java.util.concurrent.TimeoutException
-
 import cats.effect.{Blocker, ContextShift, IO, Timer}
 import sttp.client3.testing.ConvertToFuture
-
-import scala.concurrent.ExecutionContext
 import sttp.monad.MonadError
 
+import java.util.concurrent.TimeoutException
+import scala.concurrent.ExecutionContext
 import scala.concurrent.duration.DurationInt
 
 trait CatsTestBase {
+
   implicit def executionContext: ExecutionContext
 
   implicit lazy val monad: MonadError[IO] = new CatsMonadAsyncError[IO]
   implicit val contextShift: ContextShift[IO] = IO.contextShift(implicitly)
-  implicit lazy val timer: Timer[IO] = IO.timer(scala.concurrent.ExecutionContext.global)
+  implicit lazy val timer: Timer[IO] = IO.timer(implicitly)
```

Comment:
probably you can leave out the `implicitly` :)

Reply:
compiler complains, I can only replace `implicitly` with `executionContext` from above